lib/routes/routePUT.js

```diff
@@ -33,6 +33,42 @@ export default function routePUT(request, response, log, utapi) {
                     return routesUtils.responseNoBody(err, null, response, 200,
                         log);
                 });
+            } else if (request.query.versioning !== undefined) {
+                if (request.post === '') {
+                    log.debug('request xml is missing');
+                    return routesUtils.responseNoBody(
+                        errors.MalformedXML, null, response, null, log);
+                }
+                const xmlToParse = request.post;
+                return parseString(xmlToParse, (err, result) => {
+                    if (err) {
+                        log.debug('request xml is malformed');
+                        return routesUtils.responseNoBody(
+                            errors.MalformedXML, null, response, null, log);
+                    }
+                    const Status = result.VersioningConfiguration.Status ?
```

Comment:
It's AWS' format. See http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTVersioningStatus.html?shortFooter=true.